provider/plugin/plugin.go

```diff
@@ -0,0 +1,77 @@
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+type PluginProvider struct {
+	provider.BaseProvider
+	client       *http.Client
+	remoteServer string
+}
+
+func NewPluginProvider(url string) *PluginProvider {
+	return &PluginProvider{
+		client:       &http.Client{},
+		remoteServer: url,
+	}
+}
+
+// Records will make a GET call to p.remoteServer and return the results
+func (p PluginProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, error) {
+	req, err := http.NewRequest("GET", p.remoteServer, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := p.client.Do(req)
```

Comment:
you can also do `p.client.Get(p.remoteServer)`, so you don't need to do NewRequest